src/BuiltInTools/dotnet-watch/LaunchSettingsProfile.cs

```diff
@@ -45,11 +47,32 @@ public class LaunchSettingsProfile
                 return null;
             }
 
+            if (string.IsNullOrEmpty(launchProfileName))
+            {
+                // Load the default (first) launch profile
+                return ReadDefaultLaunchProfile(launchSettings, reporter);
+            }
+
+            // Load the specified launch profile
+            var namedProfile = launchSettings?.Profiles?.FirstOrDefault(f => string.Equals(f.Key , launchProfileName, StringComparison.OrdinalIgnoreCase)).Value;
```

Comment:
```suggestion
            var namedProfile = launchSettings?.Profiles?.FirstOrDefault(f => string.Equals(f.Key , launchProfileName, StringComparison.Ordinal)).Value;
```
Turns out this is case-sensitive in `dotnet run` so it should be here too.